libs/features/personalization/personalization.js

```diff
@@ -327,70 +324,70 @@ function registerInBlockActions(cmd, manifestId, targetManifestId) {
   config.mep.inBlock[blockName].commands.push(command);
 }
 
-function getSelectedElement(selector, action, rootEl) {
-  if (!selector) return null;
-  if ((action.includes('appendtosection') || action.includes('prependtosection'))) {
-    if (!selector.includes('section')) return null;
-    const section = selector.trim().replace('section', '');
-    if (section !== '' && Number.isNaN(section)) return null;
-  }
-  if (checkSelectorType(selector) === 'fragment') {
+function getSelectedElement({ selector, action, rootEl }) {
+  let modifiedSelector = selector;
+  if (!modifiedSelector) return null;
+
+  if (checkSelectorType(modifiedSelector) === 'fragment') {
     try {
-      const fragment = document.querySelector(`a[href*="${normalizePath(selector, false)}"], a[href*="${normalizePath(selector, true)}"]`);
+      const fragment = document.querySelector(`a[href*="${normalizePath(modifiedSelector, false)}"], a[href*="${normalizePath(modifiedSelector, true)}"]`);
       if (fragment) return fragment.parentNode;
       return null;
     } catch (e) {
       return null;
     }
-  }
-
-  let selectedEl;
-  if (selector.includes('.') || !['section', 'block', 'row'].some((s) => selector.includes(s))) {
-    selectedEl = querySelector(rootEl, selector);
-    if (selectedEl) return selectedEl;
-  }
-
-  const terms = selector.split(/\s+/);
+  } else {
+    MILO_BLOCKS.forEach((block) => {
+      const regex = new RegExp(`(\\s|^)(${block})\\.?(\\d+)?(\\s|$)`, 'g');
+      const match = regex.exec(modifiedSelector);
+      if (match?.length) {
+        const simplifiedSelector = match[0].replace(/\s+/g, '');
+        const n = simplifiedSelector.match(/\d+/g) || '1';
+        const cleanClassSelector = match[2];
+        const cssOptimizedSelector = ` .${cleanClassSelector}:nth-child(${n} of .${cleanClassSelector})`;
+        modifiedSelector = modifiedSelector.replace(simplifiedSelector, cssOptimizedSelector);
+      }
+    });
```

Comment:
Instead of checking for every milo block, we should assume that any selector that starts with an alpha character, and is not an html element, can be assumed to be a block and  have a `.` prepended.  

Something like this (not tested):
```
const areaSelectors = ['section', 'row', 'col'];
const actionSelectors = {
	  'primary-cta': 'p strong a',
	  'secondary-cta': 'p em a',
	  'action-area': 'p:has(em a, strong a)'
	};
const htmlEls = ['div', 'p', 'strong', 'em', 'h1', 'h2', ...];
const notBlocks = [...areaSelectors, ...Object.keys(actionSelectors), ...htmlEls];
const terms = modifiedSelector.split(/\s+/);
modifiedSelector = terms.reduce((selStr, term) => {
  const { blockName, blockIndex } = getBlockName(term);
  if (notBlocks.includes(blockName)) {
    return `${selStr} ${term}`;
  }
  return `${selStr} .${blockName}:nth-child(${blockIndex} of .${blockName})`; 
}, '');
```

Reply:
NOTE: Looks like using `.block:nth-child(4 of .block)` does not work in all circumstances.   On the example page, try using `.text4` and it will not find the element

Reply:
spoke on Slack, this is outdated

Reply:
spoke on Slack, this is outdated